### PR TITLE
CDRIVER-4563 do not create or drop `eccCollection`

### PR DESCRIPTION
**Pull request by kevinAlbs**
# Summary
- Do not create or drop `eccCollection`.
- Update `fle2v2` spec tests.
- Return error if attempting to create a QE collection with server version < 7.0.0 (maxWireVersion < 21)

Verified with [this patch build](https://spruce.mongodb.com/version/643410d857e85aedd51f7d95).

# Background & Motivation

See DRIVERS-2524 for additional background behind these changes.

### CODE REVIEW

File: src/libmongoc/src/mongoc/mongoc-database.c [hunk 1]
```diff
@@ -1100,15 +1098,37 @@ create_collection_with_encryptedFields (mongoc_database_t *database,
...
+      if (!stream) {
+         goto fail;
+      }
+      if (stream->sd->max_wire_version < WIRE_VERSION_7_0) {
+         bson_set_error (
+            error,
+            MONGOC_ERROR_PROTOCOL,
+            MONGOC_ERROR_PROTOCOL_BAD_WIRE_VERSION,
+            "Driver support of Queryable Encryption is incompatible "
+            "with server. Upgrade server to use Queryable Encryption.");
```
Comment (eramongodb) [inline src/libmongoc/src/mongoc/mongoc-database.c:1124]: Suggest including both the detected max wire version and the required server version in the error message.

File: src/libmongoc/src/mongoc/mongoc-util.c [hunk 2]
```diff
@@ -310,6 +310,8 @@ _mongoc_wire_version_to_server_version (int32_t version)
       return "5.3";
    case 17:
       return "6.0";
+   case 21:
```
Comment (eramongodb) [inline src/libmongoc/src/mongoc/mongoc-util.c:313]:
```suggestion
   case WIRE_VERSION_7_0:
```
May be worth considering replacing literals here with their macro equivalents.

**Review by eramongodb — APPROVED** [review]
Minor feedback; otherwise, LGTM.

**Review by vector-of-bool — APPROVED** [review]
LGTM

File: src/libmongoc/src/mongoc/mongoc-database.c [hunk 3]
```diff
@@ -1121,7 +1121,10 @@ create_collection_with_encryptedFields (mongoc_database_t *database,
             MONGOC_ERROR_PROTOCOL,
             MONGOC_ERROR_PROTOCOL_BAD_WIRE_VERSION,
             "Driver support of Queryable Encryption is incompatible "
-            "with server. Upgrade server to use Queryable Encryption.");
+            "with server. Upgrade server to use Queryable Encryption."
```
Comment (vector-of-bool) [inline src/libmongoc/src/mongoc/mongoc-database.c:1124]: Missing a space:

```suggestion
            "with server. Upgrade server to use Queryable Encryption. "
```

**Review by vector-of-bool — APPROVED** [review]

File: src/libmongoc/src/mongoc/mongoc-database.c [hunk 4]
```diff
@@ -1100,15 +1098,40 @@ create_collection_with_encryptedFields (mongoc_database_t *database,
...
-      create_encField_state_collection (
-         database, encryptedFields, name, "ecc", error) &&
       create_encField_state_collection (
          database, encryptedFields, name, "ecoc", error);
    if (!state_collections_ok) {
       // Failed to create one or more state collections
       goto fail;
    }
 
+   // Check the wire version to ensure server is 7.0.0 or newer.
```
Comment (jmikola) [inline src/libmongoc/src/mongoc/mongoc-database.c:1108]: This wire version check should be performed _before_ creating the QEv2 state collections. I opened [CDRIVER-4653](https://jira.mongodb.org/browse/CDRIVER-4653) to track this.

**Pull request merged**